### PR TITLE
[BUGFIX] TracingGanttChart: calculate start and end time from all spans

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
# Description

In some cases, the trace start and end time does not match the start and end time of the root span (in other words, some spans are in progress or started after the root span has ended).

* compute and memoize start and end time of the trace by scanning all spans of the trace
* introduce a GanttTrace data type which holds these pre-computed fields
* update all React props `rootSpan` --> `trace` and all references `rootSpan.startTimeUnixMs` --> `trace.startTimeUnixMs` (and `.endTime`)

# Screenshots

The trace contains spans starting after the root span has ended:
image

# Checklist

- [X] Pull request has a descriptive title and context useful to a reviewer.
- [X] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [X] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [X] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [X] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [ ] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/panels-plugin/src/test/get-trace-data-jaeger.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 The Perses Authors
```
Comment (celian-garcia) [inline ui/panels-plugin/src/test/get-trace-data-jaeger.ts:1]:
```suggestion
// Copyright 2024 The Perses Authors
```

**Review by celian-garcia — COMMENTED** [review]
Just a nit. 
Apart from that I went through the files and didn' t see anything shocking. 

I was wondering if it couldn' t be considered as a breaking change to modify the TracingGanttChart component interface but I guess one should use the TracingGanttCharPanel component right? On which the interface didn' t change.

And by the way that' s cool as I think we have some slight time issues with our spans and sometimes they are no completely align. I guess it will fix som display for us as well.

**Comment by andreasgerstmayr** [comment]
> I was wondering if it couldn' t be considered as a breaking change to modify the TracingGanttChart component interface but I guess one should use the TracingGanttCharPanel component right? On which the interface didn' t change.

Yes, `TracingGanttChart.PanelComponent` should be used.
In the `index.ts` file we only export types from `TracingGanttChart.ts`, so I'd consider everything which is not exported as an internal implementation detail.

> And by the way that' s cool as I think we have some slight time issues with our spans and sometimes they are no completely aligned. I guess it will fix som display for us as well.

:+1: let me know if you find other issues :)

**Review by celian-garcia — APPROVED** [review]
This freaking DCO on suggestion.. Sorry you'll have to fix it manually I believe... 
Otherwise LGTM

**Pull request merged**